core/blockchain.go

```diff
@@ -222,13 +219,10 @@ func (bc *BlockChain) SetHead(head uint64) {
 	bc.mu.Lock()
 	defer bc.mu.Unlock()
 
+	bc.hc.SetHead(head)
```

Comment:
There may be a very short and elegant solution to this, lets discuss after the standup.